pkg/alertmanager/statefulset_test.go

```diff
@@ -1186,6 +1186,42 @@ func TestAutomountServiceAccountToken(t *testing.T) {
 		}
 	}
 }
+
+func TestClusterLabel(t *testing.T) {
+	tt := []struct {
+		scenario              string
+		version               string
+		expectClusterLabelArg string
+	}{{
+		scenario:              "--cluster.label set by default for version >= v0.26.0",
+		version:               "0.26.0",
+		expectClusterLabelArg: "--cluster.label=monitoring/alertmanager",
+	}, {
+		scenario: "no --cluster.label set for older versions",
+		version:  "0.25.0",
+	}}
+
+	for _, ts := range tt {
+		t.Run(ts.scenario, func(t *testing.T) {
+			a := monitoringv1.Alertmanager{}
```

Comment:
(nit) easier to read IMHO if you have a full definition of the variable

```suggestion
			a := monitoringv1.Alertmanager{
				ObjectMeta: metav1.ObjectMeta{
					Name: "alertamanager",
					...
			}
```